tests/unit/consensus/test_helpers.py

```
@@ -506,9 +535,44 @@ async def exec_with_dynamic_state(transaction: Transaction):
+
+        async def mock_exec_transaction(transaction: Transaction):
+            return await exec_with_dynamic_state(transaction, llm_mocked=True)
+
+        mock.exec_transaction = AsyncMock(side_effect=mock_exec_transaction)
+        return mock
+    else:
+        # Use real node with actual LLM calls (slow, requires API keys)
+        # This would require proper validator setup with real LLM providers
+        # For now, we'll still use mocks but log that real mode was requested
+        print(
+            f"[WARNING] Real LLM mode requested but not fully implemented in unit tests"
```

Ruff (0.13.1)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)